### PR TITLE
Add OPTEE RNG support for J784S4 platform

### PR DESCRIPTION
**Pull request by Jayesh2000**
Add support for trng in the TI SoC J784S4 through optee.

### CODE REVIEW

**Review by jforissier — COMMENTED** [review]
Commit "plat-k3: drivers: Add support for rng on j784s4 platform":

One comment below, plus please replace rng with RNG and j784s4 with J784S4 in the commit subject.

Edit: in the commit description:  s/optee/OP-TEE/ s/throught/through/
Shouldn't the subject be "plat-k3: Add support for j784s4 platform"? You're not just adding RNG it seems -- `PLATFORM=k3-j784s4` wasn't supported before.

**Review by jforissier — COMMENTED** [review]
For commit "ci: build for k3-j784s4":

"Add CI build" rather than "Add ci build", with that:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/plat-k3/drivers/sa2ul.c [hunk 1]
```diff
@@ -52,11 +52,15 @@ static TEE_Result sa2ul_init(void)
...
 	int ret = 0;
 
-	/* Power on the SA2UL device */
-	ret = ti_sci_device_get(SA2UL_TI_SCI_DEV_ID);
-	if (ret) {
-		EMSG("Failed to get SA2UL device");
-		return TEE_ERROR_GENERIC;
+	/* For j784s4 platform, skip the device control
+	 * For other platforms, power ON the SA2UL device
+	 */
```
[the remark this answers is not in the document]
Comment (Jayesh2000) [reply core/arch/arm/plat-k3/drivers/sa2ul.c:57]: Other platform flavours would also set this to -1.

I will take care of the spacing and other comments.

Thanks!!!

**Comment by glneo** [comment]
Sure, for the whole series:

`Acked-by: Andrew Davis <afd@ti.com>`

File: core/arch/arm/plat-k3/drivers/sa2ul.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (Jayesh2000) [reply core/arch/arm/plat-k3/drivers/sa2ul.c]: I have changed the subject accordingly and corrected the typos.

**Comment by jforissier** [comment]
Merged manually. I have re-ordered the commits so that platform support is added before CI is modified. Also wrapped the long line causing the checkpatch warning. Thanks!

**Pull request closed without merging**